src/types/bindings/orx_types_bindings.ml

```diff
@@ -697,14 +703,59 @@ module Bindings (F : Ctypes.TYPE) = struct
     end
   end
 
+  module Anim_event = struct
+	type t =
+		| Start
+		| Stop
+		| Cut
+		| Loop
+		| Update
+		| Custom_event
+
+	let make tag = F.constant ("orxANIM_EVENT_" ^ tag) F.int64_t
+    let start = make "START"
+    let stop = make "STOP"
+    let cut = make "CUT"
+    let loop = make "LOOP"
+    let update = make "UPDATE"
+    let custom_event = make "CUSTOM_EVENT"
+
+	let map_to_constant =
+      [
+        (Start, start);
+        (Stop, stop);
+        (Cut, cut);
+        (Loop, loop);
+        (Update, update);
+        (Custom_event, custom_event);
+      ]
+
+	let map_from_constant = swap_tuple_list map_to_constant
+
+	let t =
+      F.enum "__orxANIM_EVENT_t" map_to_constant ~unexpected:(fun i ->
+          Fmt.invalid_arg "unsupported time line event type enum: %Ld" i
+      )
+
+	module Payload = struct
+      type t
+
+      let t : t structure = F.structure "__orxANIM_EVENT_PAYLOAD_t"
+      let animation = F.field t "pstAnim" (F.ptr Animation.t)
+      let name = F.field t "zAnimName" F.string
+	  (* TODO: I don't know if we have to do something special to access union members here *)
+      let () = F.seal t
```

Comment:
I _think_ we can leave off the `let () = F.seal t` until we have the full structure mapped. Then, in the `04_Anim.c` port, leave out the missing fields from the logs until after the union field mapping has been addressed.

Once we have the other union fields mapped - probably after seeing if we can fix this by adding field names upstream? - the missing log info can be added back in and the structure `seal`ed.